examples/specializecyclic.metta

```diff
@@ -0,0 +1,15 @@
+(= (f1 $f $a) (if (< $a 0)
```

Comment:
Tests whether it works for cyclic dependencies while noticing whether the function is used or not used.